crates/zk-por-core/src/merkle_sum_prover/prover.rs

```diff
@@ -92,6 +93,32 @@ impl MerkleSumTreeProver {
             }
         }
     }
+
+    pub fn prove_with_circuit(
+        &self,
+        circuit_data: &CircuitData<F, C, D>,
+        account_targets: Vec<AccountTargets>,
+    ) -> Result<ProofWithPublicInputs<F, C, D>> {
+        if account_targets.len() != self.accounts.len() {
+            return Err(anyhow::anyhow!("Account targets length does not match accounts length"));
+        }
+
+        let mut pw = PartialWitness::new();
+
+        let CircuitData { prover_only, common, verifier_only: _ } = circuit_data;
+
+        for i in 0..self.accounts.len() {
+            account_targets[i].set_account_targets(self.accounts.get(i).unwrap(), &mut pw);
+        }
+
+        let mut timing = TimingTree::new("prove_merkle_sum_tree", Level::Debug);
```

Comment:
Level:Debug should not be used in release mode; can cfg that

Reply:
can use
    #[cfg(debug_assertions)]
    let mut timing = TimingTree::new("prove_N_subproofs", log::Level::Debug);
    #[cfg(not(debug_assertions))]
    let mut timing = TimingTree::new("prove_N_subproofs", log::Level::Info);